### PR TITLE
Quaternions for sky map pointing 

### PR DESCRIPTION
**Pull request by arahlin**
More efficient conversion to/from map pixels using quaternions.  Expect a 25% speedup for non-cylindrical flat sky projections in particular (tested with MapTODMasker on an OSX machine).

All methods of G3SkyMap and subclasses that perform some conversion between pixels and angles now also have counterparts that take inputs or produce outputs that are quaternions instead of alpha/delta angles.  These methods include `PixelToQuat` and `QuatToPixel` as analogs to `PixelToAngle` and `AngleToPixel`; `GetRebinQuats` as an analog to `GetRebinAngles`; and quaternion versions of `QueryDisc` and `QueryAlphaEllipse`.

All of the mapmaking pipeline modules, e.g. `MapTODMasker` and `MapBinner`, now use the function `get_detector_pointing_pixels` (in place of `get_detector_pointing` with `AngleToPixel`) to speed up the pixel pointing calculation for each channel.

The functions for generating pointing transformation quaternions have also been reorganized.  The main functions for constructing these quaternions are now `get_origin_rotator_timestream` and `get_boresight_rotator_timestream`.  Deprecated functions that are no longer used anywhere have also been removed.

### CODE REVIEW

**Comment by nwhitehorn** [comment]
Could you provide some more commentary here on the API changes associated with this PR?

**Comment by arahlin** [comment]
> Could you provide some more commentary here on the API changes associated with this PR?

I've updated the PR description above with some more detail.

**Comment by arahlin** [comment]
Tested that this code produces numerically identical maps for the SPT3G winter field in ProjZEA.

File: maps/README.rst [hunk 1]
```diff
@@ -151,3 +151,23 @@ Small patches can be extracted from and inserted into larger flat sky maps using
...
+
+Detector Pointing
+-----------------
+
+Detector pointing timestreams are constructed by first using the ``offsets_to_quat`` function to construct the detector offset quaternion in boresight coordinates, then rotating that pointing quaternion onto the sky by applying a rotation quaternion constructed from the boresight pointing timestreams.  This is done internally for each detector in each of the mapmaking pipeline modules (``MapBinner``, ``MapMockObserver``, etc), which all require an input ``BolometerPropertiesMap`` object with offsets for each detector, and pre-computed timestreams of boresight rotation quaternions associated with each input ``Scan`` frame.
+
```
Comment (nwhitehorn) [inline maps/README.rst:173]: Thank you so much for writing this.

**Review by nwhitehorn — APPROVED** [review]

**Pull request merged**